zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1291,7 +1316,8 @@ private void runParagraph(NotebookSocket conn, HashSet<String> userAndRoles, Not
     boolean isTheLastParagraph = note.isLastParagraph(p.getId());
     if (!(text.trim().equals(p.getMagic()) || Strings.isNullOrEmpty(text)) &&
         isTheLastParagraph) {
-      note.addParagraph();
+      Paragraph newPara = note.addParagraph();
+      broadcastNewParagraph(note, newPara);
```

Comment:
The notebook was broadacasted in the following line earlier
https://github.com/apache/zeppelin/blob/master/zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java#L1634
ie when the status changes to `PENDING` (the fiirst stae change when the paragraph is run) @khalidhuseynov